.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,63 @@
+name: CI-Checks
+on: 
+  push:
+    branches-ignore:
+      -master
+      -release-*
```

Comment:
We probably only want to run towncrier on branches that are part of a pull request (as towncrier requires specifying a PR #). This can be achieved in GitHub Actions by specifying:

```yaml
on:
  pull_request:
```

We may want to run other CI tasks just when pushing branches to the repo, even if they aren't part of a visible PR yet though. For that reason, it might be best to split the changelog check into one GitHub actions pipeline, and have the other checks in a separate pipeline file, with a broader `on` definition.

Reply:
Thanks for the helpful feedback! I updated the PR with this change, there are now two Github Actions file, one with the changelog check and one with the other tests.